Constraint syntax (#1677)
dftbplus/dftbplus: doc/dftb+/manual/dftbp.tex

ground state of the system in a (carefully chosen) external
potential~\cite{Hourahine2010}.

\begin{ptable}
  \kw{Constraints} & p & & \cb & \\
  \kw{Optimiser} & m & & FIRE & \\
  \kw{ConstrTolerance} & r & & 1e-8 & \\
  \kw{MaxConstrIterations} & i & & 100 & \\
  \kw{ConvergentConstrOnly} & l & & \is{Yes} & \\
\end{ptable}

\begin{description}
\item[\is{Constraints}] Specifies the constraint(s) to be applied. Currently
  these are only implemented for atomic Mulliken populations:
  \begin{ptable}
    \kw{MullikenPopulations} & p &  & \cb & \\
  \end{ptable}
    \begin{description}
      \item[\is{MullikenPopulation}] Defines a constraint based on Mulliken
      population.
    \begin{ptable}
      \kw{Atoms} & (i|s)+ &  &  & \\
      \kw{Populations} & r &  &  & \\
      \kw{TotalPopulation} & r &  &  & \\
      \kw{Charges} & r &  &  & \\
      \kw{TotalCharge} & r &  &  & \\
    \end{ptable}
      \begin{description}
      \item[\is{Atoms}] Indices of the atoms within a constrained region. The atoms can be specified
        via index selection expressions, as described in appendix \ref{sec:index_selection}.

>       \item[\is{Populations}] Constraint on the Mulliken population on individual atoms. Must

Reviewer: ```suggestion
      \item[\is{Populations}] Constraint on the Mulliken population of individual atoms. Must
```

        contain as many entries as selected via the \is{Atoms} keyword. This keyword is mutually
        exclusive with the keywords \is{TotalPopulation}, \is{Charges} and \is{TotalCharge}.

>       \item[\is{TotalPopulation}] Sets constraint on the Mulliken population summed over all atoms

Reviewer: ```suggestion
      \item[\is{TotalPopulation}] Sets constraint on the sum of the Mulliken population for the atoms
```

        selected via the \is{Atoms} keyword. This keyword is mutually exclusive with the keywords
        \is{Population}, \is{Charges} and \is{TotalCharge}.

>       \item[\is{Charges}] Constraint on the gross charges on individual atoms. Must contain as many

Reviewer: ```suggestion
      \item[\is{Charges}] Constraint on the Mulliken (gross) charges of individual atoms. Must contain as many
```

        entries as selected via the \is{Atoms} keyword. This keyword is mutually exclusive with the
        keywords \is{Populations}, \is{TotalPopulation} and \is{TotalCharge}.

>       \item[\is{TotalPopulation}] Sets constraint on the grossed charge summed over all atoms selected

Reviewer: ```suggestion
      \item[\is{TotalCharge}] Sets constraint on the sum of the  Mulliken charges of all atoms selected
```
